### PR TITLE
WIP Qualification DDI plot infrastructure

### PR DESCRIPTION
**Pull request by abdullahhamadeh**

### CODE REVIEW

**Comment by Yuri05** [comment]
@abdullahhamadeh It seems that this PR can be closed in favor of #543?

**Comment by msevestre** [comment]
@abdullahhamadeh Or should we merge this one first and then you rebase/merge #543 on develop?

**Comment by abdullahhamadeh** [comment]
I prefer we merge this one first so the parallelization work can be reviewed more clearly

**Comment by msevestre** [comment]
ok reviewing now and merging asap

File: R/error-checks.R [hunk 1]
```diff
@@ -85,6 +85,14 @@ validateIsPositive <- function(object, nullAllowed = FALSE) {
   }
 }
 
+validateIsNonNegative <- function(object, nullAllowed = FALSE) {
+  validateIsOfType(object, c("numeric", "integer"), nullAllowed)
```
Comment (msevestre) [inline R/error-checks.R:89]: I am going to add my commnents from previous here
We have a validateIsNumeric
And I would expect StrictPositive as opposed to NonNegative..but maybe that's just the optimistic guy in me

File: R/qualification-ddi.R [hunk 2]
```diff
@@ -0,0 +1,344 @@
+#' @title getQualificationDDIPlotData
+#' @description Build dataframes and metadata for each DDI plot
+#' @param configurationPlan The configuration plan of a Qualification workflow read from json file.
+#' @return  plotDDIdata, a list of lists of the form list(dataframe,metadata) specific to each DID plot
+getQualificationDDIPlotData <- function(configurationPlan){
```
Comment (msevestre) [inline R/qualification-ddi.R:5]: can you run the styler in this file. It feels that some ident is off

File: R/qualification-ddi.R [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+                                               values = plotComponent$StartTime,
+                                               unit = plotComponent$TimeUnit)
+
+            endTime <- ospsuite::toBaseUnit(quantityOrDimension = ospDimensions$Time,
+                                            values = plotComponent$EndTime,
+                                            unit = plotComponent$TimeUnit)
+
+            pkParameterName <- generateDDIPlotPKParameterName(pkParameter,startTime,endTime)
+
+            simulationResultsFile <- configurationPlan$getSimulationResultsPath(project = projectName,
```
Comment (msevestre) [inline R/qualification-ddi.R:69]: sometime project is on the same line, sometime on the next line... I don't really care which but let's be consistent

File: R/qualification-ddi.R [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+#' @param dataframe data.frame
+#' @param metadata meta data on `data`
+#' @return dataframe for plotting goodness of fit of predictedVsObserved type
+buildQualificationDDIPredictedVsObserved <- function(dataframe,
+                                                     metadata) {
+  axesSettings <- metadata$axesSettings$predictedVsObserved
+  axesSettings$X$label <- plotDDIXLabel$predictedVsObserved
+  axesSettings$Y$label <- plotDDIYLabel$predictedVsObserved
+
+  xUnit <- axesSettings$X$unit
```
Comment (msevestre) [inline R/qualification-ddi.R:124]: where are all thos variables used?

**Review by msevestre — COMMENTED** [review]
looks good to me. Just a few very minor comments

**Comment by msevestre** [comment]
@abdullahhamadeh MErging this. Please look at my comments and we can review. Can you rebase/merge the other PR on this?

**Pull request merged**